### PR TITLE
Display more user friendly error on deleted space

### PR DESCRIPTION
**Pull request by TheOneRing**
Fixes: #9700

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-oc10</strong> https://drone.owncloud.com/owncloud/client/14656/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/14656/oc10/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/14656/oc10/guiReportUpload/serverlog.log

Screenshots:
- [logging_out.png](https://cache.owncloud.com/public/owncloud/client/14656/oc10/guiReportUpload/screenshots/logging_out.png)
- [login_with_oauth2_enabled.png](https://cache.owncloud.com/public/owncloud/client/14656/oc10/guiReportUpload/screenshots/login_with_oauth2_enabled.png)
- [sharee_tries_to_delete_shared_file_and_folder_without_permissions.png](https://cache.owncloud.com/public/owncloud/client/14656/oc10/guiReportUpload/screenshots/sharee_tries_to_delete_shared_file_and_folder_without_permissions.png)

</details>

File: src/libsync/discovery.cpp [hunk 1]
```diff
@@ -1417,23 +1417,30 @@ DiscoverySingleDirectoryJob *ProcessDirectoryJob::startAsyncServerQuery()
...
-                // The server usually replies with the custom "503 Storage not available"
-                // if some path is temporarily unavailable. But in some cases a standard 503
-                // is returned too. Thus we can't distinguish the two and will treat any
-                // 503 as request to ignore the folder. See #3113 #2884.
-                // Similarly, the server might also return 404 or 50x in case of bugs. #7199 #7586
-                _dirItem->_instruction = CSYNC_INSTRUCTION_IGNORE;
-                _dirItem->_errorString = results.error().message;
-                emit this->finished();
+            if (serverJob->isRootPath()) {
+                if (code == 404 && _discoveryData->isSpace()) {
```
Comment (fmoc) [inline src/libsync/discovery.cpp:1421]: Why is `404` equivalent to "currently"? It says "not found" which translates to "you, the client, made a mistake by querying this URL that does not exist". The resource may not come back even. "Gone` would be `410`. `50x` (`503` specifically) would be "temporarily unavailable"...

Perhaps we should just remove the word "currently".

**Review by fmoc — CHANGES_REQUESTED** [review]

File: src/libsync/discovery.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/libsync/discovery.cpp:1421]: See the discussion in the issue.
It really means currently maybe for ever, no one knows.

Comment (fmoc) [reply src/libsync/discovery.cpp:1421]: Fine by me then.

**Review by fmoc — DISMISSED** [review]

File: src/libsync/discoveryphase.cpp [hunk 2]
```diff
@@ -246,6 +246,11 @@ void DiscoveryPhase::scheduleMoreJobs()
     }
 }
 
+bool DiscoveryPhase::isSpace() const
+{
+    return !(Utility::urlEqual(_account->davUrl(), _baseUrl) || _account->davUrl().isParentOf(_baseUrl));
```
Comment (TheOneRing) [inline src/libsync/discoveryphase.cpp:251]: Please have another look, it returned true before in the case that both strings where equal

Comment (fmoc) [reply src/libsync/discoveryphase.cpp:251]: Seems okay now.

**Review by fmoc — APPROVED** [review]

**Pull request merged**